### PR TITLE
refactor: add folders and arrange files

### PR DESCRIPTION
**Pull request by sonyakim-dev**
## Summary
arranged files in folders
image

### What was changed?
- file arrangement (see above)
- changed the word 'class' to 'course'
- set base URL as 'src'

### Why were these changes made?

- Project structure makes maintaining and developing easier

## Testing

N/A

### CODE REVIEW

File: src/App.tsx
Comment (jason2020) [inline src/App.tsx:3]: Nice refactor, I like this

File: src/layouts/DarsSideBar/index.ts
Comment (jason2020) [inline src/layouts/DarsSideBar/index.ts:1]: Wait just to clarify, so the right panel is for Dars + CourseDetails?

File: src/pages/CourseGraph/style.scss
Comment (jason2020) [inline src/pages/CourseGraph/style.scss:3]: This line is breaking the CI, either disable the linting for this line, change the linting rules if you think that's better, or adjust the class name. I think to begin with an id makes more sense anyways

**Review by jason2020 — CHANGES_REQUESTED** [review]
Great work, I like this refactor. It makes sense and the layout is clear to me. Just a minor question and linting issue breaking the CI. Also, this PR conflicts with #7, but no worries, let's merge yours in first and I can fix mine (cherry pick blah blah) over the new structure.

**Comment by jason2020** [comment]
Oh right, depending on if yarn gets approved you can get rid of the package-lock. It's bad practice to have both since it leads to dependency issues. The Github actions can be fixed at a later time.

File: src/layouts/DarsSideBar/index.ts
Comment (sonyakim-dev) [reply src/layouts/DarsSideBar/index.ts:1]: @jason2020 I tried to import CourseDetail into DarsSideBar (so CourseDetail is a component of DarsSideBar); however, CourseDetail needs state props which is also used in CourseGraph. I think this state should be moved to context provider in the future.

Comment (sonyakim-dev) [reply src/layouts/DarsSideBar/index.ts:1]: I just wanted to focus on rearranging files for this pr, so technically this DarsSideBar file is not imported anymore 😟

Comment (sonyakim-dev) [reply src/layouts/DarsSideBar/index.ts:1]: ooo ok so the left panel is dars and right side bar is CourseDetail?

Comment (sonyakim-dev) [reply src/layouts/DarsSideBar/index.ts:1]: i thought the right side bar is dars cuz it's id is "dars"

Comment (jason2020) [reply src/layouts/DarsSideBar/index.ts:1]: sgsg, yea no worries abt this for now

**Review by jason2020 — APPROVED** [review]
LGTM, great work! Also imo i want to bring yarn in a future PR (after we get team approval). npm slow

**Comment by justinsheu** [comment]
Everything lgtm but Ollie and I were thinking the "layouts" folder can be changed to "panels" and the "pages" folder can be changed to "views" just to make things more clear.

**Review by olpai — APPROVED** [review]
looks good!

**Pull request merged**
